src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -160,6 +189,30 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
         }
     }
 
+    /**
+     * Checks if the max workflows limit has been reachesd
+     *  @param requestTimeOut request time out
+     *  @param maxWorkflow max workflows
+     *  @param internalListener listener for search request
+     */
+    protected void checkMaxWorkflows(TimeValue requestTimeOut, Integer maxWorkflow, ActionListener<Boolean> internalListener) {
+        QueryBuilder query = QueryBuilders.matchAllQuery();
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().query(query).size(0).timeout(requestTimeOut);
+
+        SearchRequest searchRequest = new SearchRequest(CommonValue.GLOBAL_CONTEXT_INDEX).source(searchSourceBuilder);
```

Comment:
Should we distinguish between 1000 workflows vs 1000 provisioned workflows. I am okay with just doing workflows too since 1000 its a pretty big number with a max of 10000. But we could potentially differentiate in the future if want too.

Reply:
Are you saying we should throttle based on the number of workflow steps too?

Reply:
If works fail provisioning, we aren't deleting them right now. I don't know if that is the plan

I didn't mean number of steps, I mean for example when we have a front end, users might create a bunch of workflows but not provision just for trying things out. So we can technically have a max specific either for workflows in total or workflows that are provisioned and have resources created and etc

Reply:
I see. We can definitely cover this up in the future when we will have our frontend.